### PR TITLE
test(dm): fix flaky dm 155 case test

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/5729

### What is changed and how it works?
Flaky test is triggered when source2's worker receives conflict ddls first and get stuck into waiting for new row event.
Fix: 
Make sure source1's ddl is handled first to make sure when source2 received new ddl, it can directly redirect other workers.

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: dm/tests/shardddl4_1/run.sh [hunk 1]
```diff
@@ -970,49 +970,78 @@ function DM_155_CASE {
...
+		grep 'worker' | awk -F: '{print $2}' | cut -d'"' -f 2)
+	check_log_contain_with_retry "skip conflict ddls in optimistic shard mode.*ALTER TABLE \`${shardddl1}\`.\`${tb1}\` CHANGE COLUMN \`c\` \`b\` INT" \
+		$WORK_DIR/$source1worker/log/dm-worker.log
 	run_sql_source2 "alter table ${shardddl1}.${tb2} change c b int;"
-	run_sql_source1 "insert into ${shardddl1}.${tb1} values(28,28,28,28,28);"
-	run_sql_source2 "insert into ${shardddl1}.${tb1} values(29,29,29,29,29);" # source2 won't redirect until it receives new event
-	run_sql_source2 "insert into ${shardddl1}.${tb2} values(30,30,30);"
 
+	# make sure source2 has handled tb2 ddl and trigger redirection
+	check_log_contain_with_retry "finish to handle ddls in optimistic shard mode.*alter table ${shardddl1}.${tb2} change c b int" \
```
Comment (lance6716) [inline dm/tests/shardddl4_1/run.sh:981]: if we use variable source1worker, here we should also use source2worker.

I remember then cluster initialized, source1 should be bound to worker1, maybe we don't need these dynamic variable

Comment (lichunzhu) [reply dm/tests/shardddl4_1/run.sh:979]: addressed in https://github.com/pingcap/tiflow/pull/5868/commits/f05b84de1c554ec2edf17da4e7c3b8e20b0d963a

**Review by lance6716 — APPROVED** [review]

**Comment by lichunzhu** [comment]
/run-dm-integration-test

https://ci2.pingcap.net/blue/organizations/jenkins/dm_ghpr_integration_test/detail/dm_ghpr_integration_test/6773/pipeline

File: dm/tests/shardddl4_1/run.sh [hunk 2]
```diff
@@ -970,49 +970,72 @@ function DM_155_CASE {
 	sleep 1
 	random_restart 3
 
-	run_sql_source1 "insert into ${shardddl1}.${tb1} values(25,25,25,25,25);"
-	run_sql_source2 "insert into ${shardddl1}.${tb1} values(26,26,26,26,26);"
-	run_sql_source2 "insert into ${shardddl1}.${tb2} values(27,27,27);"
-
+	# make sure source1 has received conflict ddl and handled
+	check_log_contain_with_retry "skip conflict ddls in optimistic shard mode.*ALTER TABLE \`${shardddl1}\`.\`${tb1}\` CHANGE COLUMN \`c\` \`b\` INT" \
```
Comment (WizardXiao) [inline dm/tests/shardddl4_1/run.sh:974]: Is it still possible that worker2/mysql2 is faster and here can't get this log, maybe we need to move check to [L913](https://github.com/pingcap/tiflow/pull/5868/files#diff-7b8de43fcae4ef1438831e3f9b2c6595c95f7377bdd3d89653a40b0099baae3eL913).

File: dm/tests/shardddl4_1/run.sh [hunk 3]
```diff
@@ -970,49 +970,72 @@ function DM_155_CASE {
...
+	check_log_contain_with_retry "finish to handle ddls in optimistic shard mode.*alter table ${shardddl1}.${tb2} change c b int" \
+		$WORK_DIR/worker2/log/dm-worker.log
+	run_sql_source1 "insert into ${shardddl1}.${tb1} values(25,25,25,25,25);"
+
+	# make sure source1,2 has received and done redirection
 	check_log_contain_with_retry "finish to handle ddls in optimistic shard mode.*alter table ${shardddl1}.${tb1} change c b int" \
 		$WORK_DIR/worker1/log/dm-worker.log
+	run_sql_source2 "insert into ${shardddl1}.${tb1} values(26,26,26,26,26);"
+	run_sql_source2 "insert into ${shardddl1}.${tb2} values(27,27,27);"
 	check_log_contain_with_retry "finish to handle ddls in optimistic shard mode.*alter table ${shardddl1}.${tb1} change c b int" \
```
Comment (WizardXiao) [inline dm/tests/shardddl4_1/run.sh:988]: why this check is after dml?

File: dm/tests/shardddl4_1/run.sh (hunk 2 above)
Comment (lichunzhu) [reply dm/tests/shardddl4_1/run.sh:974]: `worker2/mysql2` has two shard tables, table2 will executed finish ddl in [L976](https://github.com/pingcap/tiflow/pull/5868/files#diff-7b8de43fcae4ef1438831e3f9b2c6595c95f7377bdd3d89653a40b0099baae3eL976) after we receive `worker1/mysql1`'s handle log here. So I think we won't have such problem.

Comment (lance6716) [reply dm/tests/shardddl4_1/run.sh:974]: ping @WizardXiao

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**